go/test/endtoend/vtgate/schematracker/sharded/st_sharded_test.go

```diff
@@ -165,6 +165,37 @@ func TestMain(m *testing.M) {
 	os.Exit(exitCode)
 }
 
+func TestNewTable(t *testing.T) {
+	ctx := context.Background()
+	conn, err := mysql.Connect(ctx, &vtParams)
+	require.NoError(t, err)
+	defer conn.Close()
+
+	shard1Params := vtParams
+	shard1Params.DbName += ":-80@master"
+	connShard1, err := mysql.Connect(ctx, &shard1Params)
+	require.NoError(t, err)
+	defer connShard1.Close()
+
+	shard2Params := vtParams
+	shard2Params.DbName += ":80-@master"
+	connShard2, err := mysql.Connect(ctx, &shard2Params)
+	require.NoError(t, err)
+	defer connShard2.Close()
+
+	_ = exec(t, conn, "create table test_table (id bigint, name varchar(100))")
+
+	time.Sleep(2 * time.Second)
+
+	assertMatches(t, conn, "select * from test_table", `[]`)
+	assertMatches(t, connShard1, "select * from test_table", `[]`)
+	assertMatches(t, connShard2, "select * from test_table", `[]`)
+
+	exec(t, conn, "drop table test_table")
+
+	time.Sleep(2 * time.Second)
```

Comment:
why this extra sleep?

Reply:
Schema tracking needs enough time to propagate the schema changes implied by `DROP TABLE`

Reply:
Reload interval is set to `0.1` second, `2` seconds is plenty and should not flake the test

Reply:
we have assertmatcheswithtimeout for that.

Reply:
The function `assertMatchesWithTimeout` delays the call to `exec` while making sure the call does not exceed a certain timeout, the goal of the `time.Sleep` here is a little different as I am trying to wait after the completion of the `exec`